todo_app/app.py

```diff
@@ -1,40 +1,249 @@
+
+    # edit task
+    @app.route('/edit/<id>', methods=['GET'])
+    @login_required
+    @user_manager.has_write_permission
+    def get_edit(id):
+        item = todo.get_task(id=id)
+        if (str(item) != ""):
+            item_info = Card(item)
+            return render_template('edit.html', task=item_info)
+        else:
+            return render_template("error.html", error="Can not retrieve task information")
+
+    @app.route('/edit/<id>', methods=['POST'])
+    @login_required
+    @user_manager.has_write_permission
+    def post_edit(id):
+        response = todo.update_task(
+            id=id,
+            name=request.form['title'],
+            desc=request.form['descarea'],
+            due=request.form['duedate'],
+            status=request.form['status']
+        )
+        if (str(response) != ""):
+            return redirect('/home')
+        else:
+            return render_template("error.html", error="Task update failed !")
+
+    # delete task
+    @app.route('/delete/<id>')
+    @login_required
+    @user_manager.has_write_permission
+    def delete(id):
+        response = todo.delete_task(id=id)
+        if str(response) != "":
+            return redirect('/home')
+        else:
+            return render_template("error.html", error="failed to delete task!")
+
+    ############# UserManagement ##########################
+    @app.route('/usermanager', methods=['GET'])  # portal
+    @login_required
+    @user_manager.has_role_admin
+    def get_user_manager():
+        user_list = []
+        items = user_manager.get_all_users()
+        for item in items:
+            user_list.append(User(item))
+        item_view_model = View(user_list)
+        return render_template('userManager.html', view_model=item_view_model)
+
+    @app.route('/edituser/<id>', methods=['GET'])  # Edit user
+    @login_required
+    @user_manager.has_role_admin
+    def get_edituser(id):
+        item = user_manager.get_user(id=id)
+        if (str(item) != ""):
+            item_info = User(item)
+            return render_template('editUser.html', user=item_info)
+        else:
+            return render_template("error.html", error="failed to obtain user info!")
+
+    @app.route('/edituser/<id>', methods=['POST'])  # Edit user
+    @login_required
+    @user_manager.has_role_admin
+    def post_edituser(id):
+        response = user_manager.update_user(
+            id=id,
+            username=request.form['username'],
+            role=request.form['role']
+        )
+        if (str(response) != ""):
+            return redirect('/usermanager')
+        else:
+            return render_template("error.html", error="failed to update user!")
+
+    @app.route('/deleteuser/<id>')  # delete user
+    @login_required
+    @user_manager.has_role_admin
+    def deleteuser(id):
+        response = user_manager.delete_user(id=id)
+        if str(response) != "":
```

Comment:
This is a bit odd. Was the result `None` when there was an error? I would have expected it to still be a DeleteResult with deleted_count equal to 0.